model/src/cfd.rs

```diff
@@ -616,6 +617,7 @@ pub struct Cfd {
     settlement_interval: Duration,
     quantity: Usd,
     counterparty_network_identity: Identity,
+    counterparty_libp2p_identity: Option<Libp2pIdentity>,
```

Comment:
❓ So as soon as all CFDs have an associated counterparty peer ID we can remove the `Option`, right?

Reply:
❓ Does this really need to be optional? You've already set this field to a default value in the database. If we load a CFD from the database, won't it already have this field included? I don't see when we can get a `None` here.

Reply:
Originally I wanted to just use the default, but if we ever log it in a context that is *not* the new protocol it will be weird. 
We could still accept to always set it and be careful when to use this field for logging.

Reply:
IMHO it's much cleaner to have it as Optional. in case you upgrade a maker with a CFD open, this field will not be set (and the lookup will fail, as we can't create a mapping on that side). I think it's much cleaner to encode such things with an Option than a magic value (in the DB it makes a bit more sense to have a default if we convert it to an Option when reading; the added value is preventing setting it to `None`, all new CFDs ought to have that field)

Reply:
So you do think this can be `None` then?

Reply:
yes, it can be `None`, in case you upgrade with an open CFD and you can't look up the counterparty id (which will be the case on the maker).